engine/gamesys/src/gamesys/components/comp_factory.cpp

```diff
@@ -445,4 +459,25 @@ namespace dmGameSystem
     }
 
 
+    dmGameObject::HInstance CompFactorySpawn(HFactoryWorld world, HFactoryComponent component, dmGameObject::HCollection collection,
+                                                uint32_t index, dmhash_t id,
+                                                const dmVMath::Point3& position, const dmVMath::Quat& rotation, const dmVMath::Vector3& scale,
+                                                dmGameObject::HPropertyContainer properties)
+    {
+        dmGameObject::HPrototype prototype = CompFactoryGetPrototype(world, component);
+        const char* path = CompFactoryGetPrototypePath(world, component);
+
+        dmGameObject::HInstance instance = dmGameObject::Spawn(collection, prototype, path, id, properties, position, rotation, scale);
+        if (instance != 0x0)
+        {
+            dmGameObject::AssignInstanceIndex(index, instance);
+        }
+        else
+        {
+            dmGameObject::ReleaseInstanceIndex(index, collection);
+        }
+        return instance;
+    }
+
+
```

Comment:
Moved from script_factory.cpp, helping us to encapsulate the functionality more.